core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -2287,6 +2287,23 @@ private[spark] object Utils extends Logging with SparkClassUtils {
     }.map(threadInfoToThreadStackTrace)
   }
 
+  /** Return a heap dump. Used to capture dumps for the web UI */
+  def getHeapHistogram(): Array[String] = {
+    // From Java 9+, we can use 'ProcessHandle.current().pid()'
+    val pid = getProcessName().split("@").head
+    val builder = new ProcessBuilder("jmap", "-histo:live", pid)
+    builder.redirectErrorStream(true)
+    val p = builder.start()
+    val r = new BufferedReader(new InputStreamReader(p.getInputStream()))
+    val rows = ArrayBuffer.empty[String]
+    var line = ""
+    while (line != null) {
+      if (line.nonEmpty) rows += line
+      line = r.readLine()
+    }
+    rows.toArray
```

Comment:
Use `IOUtils.readLines` or `Source.getLines` instead ?

Reply:
> Use `IOUtils.readLines` or `Source.getLines` instead ?

For this one, I was thinking about adding a new configuration to limit the results like Top 100 or Top 1000. I'll handle this with that new configuration together.